src/test/scala/scala/xml/AttributeTest.scala

```diff
@@ -65,4 +65,97 @@ class AttributeTest {
     assertEquals("apple", xml \@ "bar")
   }
 
-}
\ No newline at end of file
+  @Test
+  def attributePathRootNoAttribute: Unit = {
+    val xml = <foo />
+    assertEquals(NodeSeq.Empty, xml \ "@bar")
+  }
+
+  @Test(expected=classOf[IllegalArgumentException])
+  def attributePathIllegalEmptyAttribute: Unit = {
+    val xml = <foo />
+    xml \ "@"
+  }
+
+  @Test
+  def attributePathRootWithOneAttribute: Unit = {
+    val xml = <foo bar="apple" />
+    assertEquals(Group(Text("apple")), xml \ "@bar")
+    // assertEquals(NodeSeq.fromSeq(Seq(Text("apple"))), xml \ "@bar")
+  }
+
+  @Test
+  def attributePathRootWithMissingAttributes: Unit = {
+    val xml = <foo bar="apple" />
+    assertEquals(NodeSeq.Empty, xml \ "@oops")
+  }
+
+  @Test
+  def attributePathDuplicateAttribute: Unit = {
+    val xml = Elem(null, "foo",
```

Comment:
I am using xml literals where I can.  This particular unit test doesn't because you can't: The literal syntax won't let you actually create an element with duplicate attributes.  I'm trying to force it by manually constructing a `scala.xml.Elem` object.  This is a hoax though, since it just gets removed by `MetaData.normalize` in the `Elem` constructor.